javascript/ql/src/Security/CWE-078/IndirectCommandInjection.ql

```diff
@@ -0,0 +1,38 @@
+/**
+ * @name Indirect uncontrolled command line
+ * @description Using command line parameters to construct other
+ *              command line arguments may allow a malicious user to
```

Comment:
> a malicious user

I'm afraid I still don't quite understand the threat model here. When you previously explained the problem to me, it sounded like you were concerned about non-malicious users accidentally shooting themselves in the foot due to unexpected variable substitution. That seems like a valid concern, but what's the security angle again?

Reply:
The security angle is that you can not safely invoke a flagged program with the otherwise safe `execFile`, because the command line arguments in that case may come from a malicoius user.

Consider:

```js
// this is always unsafe, and will be flagged by `js/command-line-injection`
cp.exec(`${myFlaggedBinary} ${tainted}`); 

// this is the suggested solution, but it does not work because the actual command injection occurs inside `myFlaggedBinary`.
cp.execFile(myFlaggedBinary, [tainted])

```

Reply:
Thank you, that makes sense. Could you integrate this into the qhelp?